stdlib/Queue.delta

```diff
@@ -0,0 +1,75 @@
+/// First-in-first-out data structure
+class Queue<T> {
+
+    var head: mutable Node<T>*?;
+    var tail: mutable Node<T>*?;
+    var size: uint;
+
+    /// Initialize an empty Queue
+    init() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
+    /// Add to the front of the queue
+    mutating func push(value: T*) {
+        let newNode = cast<mutable Node<T>*>(malloc(sizeof(Node<T>))!);
+        *newNode ~= Node<T>(value);
+
+        if (isEmpty()) {
+            head = newNode;
+            tail = newNode;
+        } else {
+            tail!.next = newNode;
+            tail = newNode;
+        }
+        size++;
+    }
+
+    /// Retrieve and remove head of the queue
+    mutating func pop() -> T {
+        let value = *head!.value;
+        let next = head!.next;
+        free(cast<mutable void*?>(head));
+        head = next;
+
+        if (head == null) {
+            tail = null;
+        }
+
+        size--;
+        return value;
+    }
+
+    /// Access element in the front of the queue
+    func front() -> T {
```

Comment:
We want to return a pointer to the element here instead of a separate object.

Reply:
Should this be named `first()` to be consistent with Array?

Reply:
The current naming matches C++ [queue](http://www.cplusplus.com/reference/queue/queue/)

Reply:
But the naming of `Array.first()` doesn't match C++ vector which has `front()`, so I think we don't want `front()` here either.